app/components/Editor/index.js

```diff
@@ -1,137 +1,304 @@
+  UnderlineMark,
+  CodeMark
+} from './components/marks';
+import { List } from 'immutable';
+import styles from './Editor.css';
+import editList from './plugins/editList';
+import pasteLink from './plugins/pasteLink';
+import images from './plugins/images';
+import MarkdownShortcuts from './plugins/markdown';
+import { html } from './serializer';
+import schema from './schema';
 
-type State = {
-  editorState: any
-};
+//Not really needed since the schema owerwrites this.
+const initialValue = Value.fromJSON({
+  document: {
+    nodes: [
+      {
+        object: 'block',
+        type: 'paragraph',
+        nodes: [
+          {
+            object: 'text',
+            leaves: []
+          }
+        ]
+      }
+    ]
+  }
+});
 
-/**
- * Custom Editor component
- *
- * ### Example Usage
- * ```js
- * <Editor />
- * ```
- *
- *  ### Also avialable as a redux form field
- * ```js
- * <Field component={EditorField.Field} />
- * ```
+// Simply enables the user to insert tabs when editing
+// TODO switch tabs for spaces in code blocks, make backspace remove two spaces
+function insertTab(options) {
+  return {
+    onKeyDown(event: React.KeyboardEvent<any>, editor, next) {
+      if (event.key == 'Tab') {
+        event.preventDefault();
+        editor.insertText('\t');
+      } else return next();
+    }
+  };
+}
+
+/*  softEnter(Array<string>) => onKeyDown()
+ *  A plugin that enables soft enter if selection has blocks
+ *  of a type specified. Takes an array of types as its only argument
+ *  TODO consider adding soft enter on shift for paragraphs and removing the need for shift+enter in
+ *  lists
  */
-export default class Editor extends Component<Props, State> {
-  editor: ?HTMLElement;
 
+function softEnter(types) {
+  return {
+    onKeyDown(event, editor, next) {
+      if (!editor.value.blocks.some(block => types.includes(block.type))) {
+        return next();
+      }
+      if (event.key == 'Enter') {
+        // if user is holding shift, default behaviour, (new block)
+        if (event.shiftKey) return next();
+        event.preventDefault();
+        editor.insertText('\n');
+      } else return next();
+    }
+  };
+}
+
+// A set of commands and queries to edit links
+const linkCommands = {
+  commands: {
+    wrapLink(editor, url) {
+      editor.wrapInline({ data: { url: url, text: url }, type: 'link' });
+    },
+    unwrapLink(editor) {
+      editor.unwrapInline('link');
+    }
+  },
+  queries: {
+    isLinkActive(editor) {
+      editor.value.inlines.some(inline => inline.type == 'link');
+    }
+  }
+};
+
+const DEFAULT_BLOCK = 'paragraph';
+
+const plugins = [
+  editList(),
+  insertTab(),
+  softEnter(['code-block']),
+  linkCommands,
+  pasteLink(),
+  images(),
+  MarkdownShortcuts
+];
+
+export default class Editor extends React.Component<Props, State> {
   state = {
-    editorState: createEditorState(
-      convertToRaw(importer(this.props.value || ''))
-    )
+    editorValue: this.props.value
```

Comment:
What happen if the props change (aka. does this have the same issue as the old editor when changing the input value)?

Reply:
Not sure what this issue is, but I havent had any issues. It might be a good idea to not rerender when the value prop changes though.